### ISSUE TITLE
Make the code more scala-like and less java-like

### ISSUE THREAD

**Issue by TsundereBug** [TsundereBug/Scolor#2]
Right now the code is more like Java. Not enough programming that separates it from Java libs.

**Comment by sebiglesias** [comment]
@TsundereBug I'll try to solve it!

**Comment by EdgeCaseBerg** [comment]
Hello! I was looking for hacktoberfest projects to contribute to and saw this issue. One thing I noticed was that your test folder has a single main file for testing instead of using something like scalatest. 

If you're not opposed to scalatest and helping me understand what the code in the test folder should be testing for; I'm happy to open up a PR with a test added in if you'd like. Otherwise I'm just going to look around the codebase to see if there's anything I see that can be made more "scala-y" that I can spot

**Comment by EdgeCaseBerg** [comment]
Are there any files in particular you were hoping to make more scala-like? Looking around, I can see things like this (OTFArray):

```
override def length(b: ByteAllocator): UInt = elems.map(_.length(b)).toArray.qsum
```

Which could be rewritten to not use `qsum` at all and avoid creating a new sequence that gets turned into an array for use by said method by using foldLeft:

```
override def length(b: ByteAllocator): UInt = elems.foldLeft(UInt(0)) {
  	case (accum, elem) => accum + elem.length(b)
  }
```

(I'm not familiar with Spire, but I assume that this is similar functionality?). Is this something you'd consider in line with this Issue? 

Similar to the above, you can do the same with the other method in that file:

```
override def getBytes(b: ByteAllocator): Array[UByte] = elems.flatMap(_.getBytes(b)).toArray
```

Could become:

```
  override def getBytes(b: ByteAllocator): Array[UByte] = elems.foldLeft(Array.empty[UByte]) {
  	case (accum, elem) => accum ++ elem.getBytes(b)
  }
```

I can look around and make a PR for this if that fits what you're looking for

**Comment by TsundereBug** [comment]
Yes, switching to scalatest and what you show in your last comment is the type of stuff I am looking for :) Glad you're excited to help @EdgeCaseBerg

**Comment by EdgeCaseBerg** [comment]
Happy to help @TsundereBug! 
I have some plans tonight that might keep me away from my computer but I'll try to do some of the stuff I mentioned when I can tonight if possible and tomorrow if not today.

**Comment by TsundereBug** [comment]
Fixed by #7 and #9. Code is more defining and less of a java lib. It has a style now that sets it apart from a standard java lib.
